contracts/examples/bonding-curve/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+#![no_std]
+#![allow(unused_attributes)]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+mod bc_function;
+use bc_function::{BCFunction, CurveArguments, SupplyType, Token};
+
+mod linear_function;
+use linear_function::LinearFunction;
+
+const TOKEN_NUM_DECIMALS: usize = 18;
+
+#[elrond_wasm_derive::contract]
+pub trait BondingCurve {
+	#[init]
+	fn init(&self, issued_token: Token, exchanging_token: Token) {
+		self.issued_token().set(&issued_token);
+		self.exchanging_token().set(&exchanging_token);
+	}
+
+	// endpoint - owner-only
+
+	#[payable("*")]
+	#[endpoint(issueToken)]
+	fn issue_token(
+		&self,
+		initial_supply: Self::BigUint,
+		#[payment] issue_cost: Self::BigUint,
+	) -> SCResult<AsyncCall<Self::SendApi>> {
+		only_owner!(self, "only owner may call this function");
+
+		require!(
+			self.issued_token().is_empty(),
+			"wrapped egld was already issued"
+		);
+
+		let caller = self.blockchain().get_caller();
+
+		self.issue_started_event(
+			&caller,
+			&self.issued_token().get().identifier,
+			&initial_supply,
+		);
+
+		Ok(ESDTSystemSmartContractProxy::new_proxy_obj(self.send())
+			.issue_fungible(
+				issue_cost,
+				&self.issued_token().get().name,
+				&self.issued_token().get().identifier.into_boxed_bytes(),
```

Comment:
Token Identifier != Token Ticker.

Token identifier is what you get after issue, which is Token Ticker + "-" + 6 random characters.